index.d.ts

```diff
@@ -5,6 +5,39 @@ declare class TimeoutErrorClass extends Error {
 
 declare namespace pTimeout {
 	type TimeoutError = TimeoutErrorClass;
+
+	type Options = {
+		/**
+		Custom implementations for the `setTimeout` and `clearTimeout` functions.
+
+		Useful for testing purposes, in particular to work around [`sinon.useFakeTimers()`](https://sinonjs.org/releases/latest/fake-timers/).
+
+		@example
+		```
+		import pTimeout = require('p-timeout');
+		import sinon = require('sinon');
+
+		(async () => {
+			const originalSetTimeout = setTimeout;
+			const originalClearTimeout = clearTimeout;
+
+			sinon.useFakeTimers();
+
+			// Use `pTimeout` without being affected by `sinon.useFakeTimers()`:
+			await pTimeout(doSomething(), 2000, undefined, {
+				customTimers: {
+					setTimeout: originalSetTimeout,
+					clearTimeout: originalClearTimeout
+				}
+			});
+		})();
+		```
+		*/
+		readonly customTimers?: {
+			setTimeout?: typeof global.setTimeout;
+			clearTimeout?: typeof global.clearTimeout;
```

Comment:
I don't think these should be optional. I think you should either have to set both or none.